geode-assembly/src/acceptanceTest/java/org/apache/geode/launchers/ServerStartupValueRecoveryNotificationTest.java

```diff
@@ -0,0 +1,173 @@
+package org.apache.geode.launchers;
+
+import static org.apache.geode.test.awaitility.GeodeAwaitility.await;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.rules.TestName;
+
+import org.apache.geode.internal.AvailablePortHelper;
+import org.apache.geode.test.junit.rules.gfsh.GfshRule;
+
+public class ServerStartupValueRecoveryNotificationTest {
+
+
+  private static final String SERVER_1_NAME = "server1";
+  private static final String LOCATOR_NAME = "locator";
+  private static final String DISKSTORE_1 = "diskstore1";
+  private static final String DISKSTORE_2 = "diskstore2";
+
+  @Rule
+  public GfshRule gfshRule = new GfshRule();
+
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @Rule
+  public TestName testName = new TestName();
+
+  private Path locatorFolder;
+  private Path server1Folder;
+  private int locatorPort;
+  private String startServer1Command;
+
+  @Before
+  public void persistentRegionThatRequiresValueRecovery() throws IOException {
+    locatorFolder = temporaryFolder.newFolder(LOCATOR_NAME).toPath().toAbsolutePath();
+    server1Folder = temporaryFolder.newFolder(SERVER_1_NAME).toPath().toAbsolutePath();
+    Path diskStore1Folder = temporaryFolder.newFolder(DISKSTORE_1).toPath().toAbsolutePath();
+    Path diskStore2Folder = temporaryFolder.newFolder(DISKSTORE_2).toPath().toAbsolutePath();
+    int[] ports = AvailablePortHelper.getRandomAvailableTCPPorts(1);
+    locatorPort = ports[0];
+
+    Files.createDirectories(server1Folder);
+
+
+    String startLocatorCommand = String.join(" ",
+        "start locator",
+        "--name=" + LOCATOR_NAME,
+        "--dir=" + locatorFolder,
+        "--port=" + locatorPort,
+        "--locators=localhost[" + locatorPort + "]");
+
+    startServer1Command = String.join(" ",
+        "start server",
+        "--name=" + SERVER_1_NAME,
+        "--dir=" + server1Folder,
+        "--locators=localhost[" + locatorPort + "]",
+        "--disable-default-server");
+
+    String createDiskStore1 = String.join(" ",
```

Comment:
If you're going to String.join here, you can split the command into multiple lines to make it more readable. If you're not going to split the string, you can remove the String.join. (Same comment for the next createDiskStore command.)

Reply:
Fixed